src/test/java/com/pillarglobal/miniApp/services/CalculatorServiceImplTest.java

```diff
@@ -0,0 +1,23 @@
+package com.pillarglobal.miniApp.services;
+
+import com.pillarglobal.miniApp.Calculator.controllers.services.CalculatorServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+
+class CalculatorServiceImplTest {
+
+    @InjectMocks
+    private CalculatorServiceImpl calculatorService;
+
+    @Test
+    void calculateExpression() {
+    }
+
+    @Test
+    void hasPrecedence() {
+    }
+
+    @Test
+    void applyOperations() {
+    }
```

Comment:
Add tests to be sure parsing works

Reply:
Test are on the way right know in the next branch/pull request.